Here's the cleanup. Even before my changes the unit tests don't work as is in my environment. I'm on a Mac and need to use the -binary flag to point gjbt to the chrome binary - maybe that's why, maybe not.  I tried running 'go test' and 'go test -binary /path/to/my/chrome' but they didn't work. The first failure reads

--- FAIL: TestSuccessNoOutput (0.01s)
	main_test.go:12: exit code; want 0; got 1

The second is the same but for another line in main_test.go.

I'm happy to leave well enough alone. I wish I did but I don't have free cycles at this time.

--- a/main_test.go
+++ b/main_test.go
@@ -26,3 +26,17 @@ func TestFail(t *testing.T) {
 	r.exitCode(1)
 	r.grepBoth("failed for no reason", "failed to find fail output")
 }
+
+func TestEnv(t *testing.T) {
+	r := testRunner(t, "test.004")
+	r.setEnv("BANANA", "banana")
+	r.run()
+	r.exitCode(0)
+}
+
+func TestFlags(t *testing.T) {
+	r := testRunner(t, "test.005")
+	r.run("-v", ".")
+	r.exitCode(0)
+	r.grepBoth("PASS: Test005", "failed to test pass")
+}

--- a/testdata/test.004/init_test.go
+++ b/testdata/test.004/init_test.go
@@ -1,0 +1,16 @@
+// +build js
+
+package main_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+func TestMain(m *testing.M) {
+	i := m.Run()
+
+	js.Global.Call("eval", fmt.Sprintf("window.$GopherJSTestResult = %v", i))
+}

--- a/testdata/test.004/js_test.go
+++ b/testdata/test.004/js_test.go
@@ -1,0 +1,19 @@
+// +build js
+
+package main_test
+
+import (
+	"os"
+	"testing"
+)
+
+const (
+	envVar = "BANANA"
+)
+
+func Test(t *testing.T) {
+	want := "banana"
+	if got := os.Getenv(envVar); got != want {
+		t.Fatalf("expected env %v to be %v; got %v", envVar, want, got)
+	}
+}

--- a/testdata/test.005/init_test.go
+++ b/testdata/test.005/init_test.go
@@ -1,0 +1,16 @@
+// +build js
+
+package main_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+func TestMain(m *testing.M) {
+	i := m.Run()
+
+	js.Global.Call("eval", fmt.Sprintf("window.$GopherJSTestResult = %v", i))
+}

--- a/testdata/test.005/js_test.go
+++ b/testdata/test.005/js_test.go
@@ -1,0 +1,10 @@
+// +build js
+
+package main_test
+
+import (
+	"testing"
+)
+
+func Test005(t *testing.T) {
+}

--- a/util_test.go
+++ b/util_test.go
@@ -47,6 +47,8 @@ type testRunnerData struct {
 
 	ran bool
 
+	env []string
+
 	actExitCode int
 	stdout      bytes.Buffer
 	stderr      bytes.Buffer
@@ -56,7 +58,12 @@ func testRunner(t *testing.T, dir string) *testRunnerData {
 	return &testRunnerData{
 		dir: dir,
 		t:   t,
+		env: os.Environ(),
 	}
+}
+
+func (tr *testRunnerData) setEnv(key, value string) {
+	tr.env = append(tr.env, key+"="+value)
 }
 
 func (tr *testRunnerData) run(flagAndArgs ...string) {
@@ -80,6 +87,7 @@ func (tr *testRunnerData) run(flagAndArgs ...string) {
 	cmd.Dir = filepath.Join(dir)
 	cmd.Stdout = &tr.stdout
 	cmd.Stderr = &tr.stderr
+	cmd.Env = tr.env
 
 	var exitCode int
 

